File: app/src/main/java/dev/ragnarok/fenrir/api/interfaces/IFriendsApi.java
package dev.ragnarok.fenrir.api.interfaces;

import androidx.annotation.CheckResult;

import java.util.List;

import dev.ragnarok.fenrir.api.model.Items;
import dev.ragnarok.fenrir.api.model.VKApiUser;
import dev.ragnarok.fenrir.api.model.VkApiFriendList;
import dev.ragnarok.fenrir.api.model.response.DeleteFriendResponse;
import dev.ragnarok.fenrir.api.model.response.OnlineFriendsResponse;
import io.reactivex.rxjava3.core.Single;


public interface IFriendsApi {

    @CheckResult
    Single<OnlineFriendsResponse> getOnline(int userId, String order, int count,
                                            int offset, String fields);

    @CheckResult
    Single<Items<VKApiUser>> get(Integer userId, String order, Integer listId, Integer count, Integer offset,
                                 String fields, String nameCase);

    @CheckResult
    Single<List<VKApiUser>> getByPhones(String phones, String fields);

    @CheckResult
    Single<Items<VKApiUser>> getRecommendations(Integer count, String fields, String nameCase);

    @CheckResult
    Single<Items<VkApiFriendList>> getLists(Integer userId, Boolean returnSystem);

    @CheckResult
    Single<DeleteFriendResponse> delete(int userId);

    @CheckResult
    Single<Integer> add(int userId, String text, Boolean follow);

    @CheckResult
    Single<Items<VKApiUser>> search(int userId, String query, String fields, String nameCase,
                                    Integer offset, Integer count);

    @CheckResult
    Single<List<VKApiUser>> getMutual(Integer sourceUid, int targetUid, int count, int offset, String fields);

}

File: app/src/main/java/dev/ragnarok/fenrir/api/interfaces/IFriendsApi.kt
package dev.ragnarok.fenrir.api.interfaces

import androidx.annotation.CheckResult
import dev.ragnarok.fenrir.api.model.Items
import dev.ragnarok.fenrir.api.model.VKApiFriendList
import dev.ragnarok.fenrir.api.model.VKApiUser
import dev.ragnarok.fenrir.api.model.response.DeleteFriendResponse
import dev.ragnarok.fenrir.api.model.response.OnlineFriendsResponse
import io.reactivex.rxjava3.core.Single

interface IFriendsApi {
    @CheckResult
    fun getOnline(
        userId: Int, order: String?, count: Int,
        offset: Int, fields: String?
    ): Single<OnlineFriendsResponse>

    @CheckResult
    operator fun get(
        userId: Int?, order: String?, listId: Int?, count: Int?, offset: Int?,
        fields: String?, nameCase: String?
    ): Single<Items<VKApiUser>>

    @CheckResult
    fun getByPhones(phones: String?, fields: String?): Single<List<VKApiUser>>

    @CheckResult
    fun getRecommendations(
        count: Int?,
        fields: String?,
        nameCase: String?
    ): Single<Items<VKApiUser>>

    @CheckResult
    fun getLists(userId: Int?, returnSystem: Boolean?): Single<Items<VKApiFriendList>>

    @CheckResult
    fun delete(userId: Int): Single<DeleteFriendResponse>

    @CheckResult
    fun add(userId: Int, text: String?, follow: Boolean?): Single<Int>

    @CheckResult
    fun search(
        userId: Int, query: String?, fields: String?, nameCase: String?,
        offset: Int?, count: Int?
    ): Single<Items<VKApiUser>>

    @CheckResult
    fun getMutual(
        sourceUid: Int?,
        targetUid: Int,
        count: Int,
        offset: Int,
        fields: String?
    ): Single<List<VKApiUser>>
}